tree/dataframe/inc/ROOT/RDFActionHelpers.hxx

```diff
@@ -246,49 +245,50 @@ extern template void
 FillHelper::Exec(unsigned int, const std::vector<unsigned int> &, const std::vector<unsigned int> &);
 
 template <typename HIST = Hist_t>
-class FillTOHelper : public RActionImpl<FillTOHelper<HIST>> {
-   std::unique_ptr<TThreadedObject<HIST>> fTo;
+class FillParHelper : public RActionImpl<FillParHelper<HIST>> {
+   std::vector<HIST*> fObjects;
 
 public:
-   FillTOHelper(FillTOHelper &&) = default;
-   FillTOHelper(const FillTOHelper &) = delete;
+   FillParHelper(FillParHelper &&) = default;
+   FillParHelper(const FillParHelper &) = delete;
 
-   FillTOHelper(const std::shared_ptr<HIST> &h, const unsigned int nSlots)
-      : fTo(std::make_unique<TThreadedObject<HIST>>(*h))
+   FillParHelper(const std::shared_ptr<HIST> &h, const unsigned int nSlots)
+      : fObjects(nSlots, nullptr)
    {
-      fTo->SetAtSlot(0, h);
+      fObjects[0] = h.get();
       // Initialise all other slots
-      for (unsigned int i = 0; i < nSlots; ++i) {
-         fTo->GetAtSlot(i);
+      for (unsigned int i = 1; i < nSlots; ++i) {
+         fObjects[i] = new HIST(*fObjects[0]);
```

Comment:
`TH1D`'s copy ctor is fairly heavy weight (see [its implementation](https://root.cern/doc/master/TH1_8cxx_source.html#l02580)), but here in principle it would be enough to create empty histograms with the correct binning, without performing a deep copy of `fObjects[0]`

Reply:
Right. How would you do it in the general case?